### PR TITLE
Delegations resolver refactor 

### PR DESCRIPTION
**Pull request by niklaslong**
Prelude to #70 and part of #52 

This refactors creating a delegation and moves business logic from the `delegations` resolver to the `delegations` context.

Notes: 
- Uses Ecto.Multi to group DB transactions: https://hexdocs.pm/ecto/Ecto.Multi.html#content
- Open question: what to validate at GraphQL layer (possibly presence of 2 emails, 2 IDs)?
   Current validation is all being done at DB level.

### CODE REVIEW

File: lib/liquid_voting_web/resolvers/delegations.ex [hunk 1]
```diff
@@ -11,59 +11,26 @@ defmodule LiquidVotingWeb.Resolvers.Delegations do
...
 
-  def create_delegation_with_valid_arguments(args) do
-    case Delegations.create_delegation(args) do
       {:error, changeset} ->
         {:error,
          message: "Could not create delegation", details: ChangesetErrors.error_details(changeset)}
 
-      {:ok, delegation} ->
-        {:ok, delegation}
+      {:error, name, changeset, _} ->
```
Comment (oliverbarnes) [inline lib/liquid_voting_web/resolvers/delegations.ex:27]: 👍

File: lib/liquid_voting/delegations.ex [hunk 2]
```diff
@@ -109,28 +113,33 @@ defmodule LiquidVoting.Delegations do
...
-      iex> create_delegation(%{field: bad_value})
-      Ecto.*Error
-
-  """
-  def create_delegation!(attrs \\ %{}) do
+  def create_delegation(attrs) do
     %Delegation{}
     |> Delegation.changeset(attrs)
-    |> Repo.insert!()
+    |> Repo.insert()
```
[the remark this answers is not in the document]
Comment (niklaslong) [reply lib/liquid_voting/delegations.ex:142]: The diff made it look weird, I'm actually not changing the behaviour. I only removed `create_delegation!` as we weren't using it at all.

**Comment by niklaslong** [comment]
Could you merge tomorrow, I won't have time to do any smoke testing tonight?

File: lib/liquid_voting/delegations.ex (hunk 2 above)
[the remark this answers is not in the document]
Comment (oliverbarnes) [reply lib/liquid_voting/delegations.ex:142]: ah, got it! Resolved then.

**Comment by oliverbarnes** [comment]
Sure thing. Thanks, dude

**Pull request merged**